docs/content/standards/kiosk-apps/nft-rental.mdx

```diff
@@ -0,0 +1,428 @@
+    cap: &KioskOwnerCap,
+    protected_tp: &ProtectedTP<T>,
+    item_id: ID,
+    duration: u64,
+    price_per_day: u64,
+    ctx: &mut TxContext)
+```
+
+The `delist` function allows the renter to delist an item, as long as it's not currently being rented. The function also places (or locks, if a lock rule is present) the object back to owner's kiosk. You should mint an empty `TransferPolicy` even if you don't want to apply any royalties.
+
+```move
+public fun delist<T: key + store>(
+		kiosk: &mut Kiosk, 
+		cap: &KioskOwnerCap, 
+		transfer_policy: &TransferPolicy<T>, 
+		item_id: ID, 
+		_ctx: &mut TxContext)
+```
+
+The `rent` function enables renting a listed `Rentable`. It permits anyone to borrow an item on behalf of another user, provided they have the `Rentables` extension installed. The `rental_policy` defines the portion of the coin that is retained as fees and added to the rental policy's balance.
+
+```move
+public fun rent<T: key + store>(
+    renter_kiosk: &mut Kiosk,
+    borrower_kiosk: &mut Kiosk,
+    rental_policy: &mut RentalPolicy<T>,
+    item_id: ID,
+    coin: Coin<SUI>,
+    clock: &Clock,
+    ctx: &mut TxContext)
+```
+
+The `borrow` function enables the borrower to acquire the `Rentable` by reference from their bag.
+
+```move
+public fun borrow<T: key + store>(
+		kiosk: &mut Kiosk, 
+		cap: &KioskOwnerCap, 
+		item_id: ID, 
+		_ctx: &mut TxContext
+): &T
+```
+
+The `borrow_val` function enables the borrower to temporarily acquire the `Rentable` with an agreement or promise to return it. The `Promise` stores all the information about the `Rentable`, facilitating the reconstruction of the `Rentable` upon object return.
+
+```move
+public fun borrow_val<T: key + store>(
+		kiosk: &mut Kiosk, 
+		cap: &KioskOwnerCap, 
+		item_id: ID, 
+		_ctx: &mut TxContext
+): (T, Promise)
+```
+
+The `return_val` function enables the borrower to return the borrowed item.
+
+```move
+public fun return_val<T: key + store>(
+		kiosk: &mut Kiosk, 
+		object: T, 
+		promise: Promise, 
+		_ctx: &mut TxContext)
+```
+:::note
+
+The `reclaim` functionality is manually invoked and the rental service provider is responsible for ensuring that the renter is reminded to `reclaim`. As such, this can cause the borrower to hold the asset for longer than the rental period. This can be mitigated through modification of the current contract by adding an assertion in the `borrow` and `borrow_val` functions to check if the rental period has expired.
+
+:::
+
+The `reclaim` function enables an owner to claim back their asset after the rental period is over and place it inside their kiosk. If a lock rule is present, the example also locks the item inside the owner kiosk. 
+
+```move
+public fun reclaim<T: key + store>(
+        renter_kiosk: &mut Kiosk, 
+        borrower_kiosk: &mut Kiosk, 
+        transfer_policy: &TransferPolicy<T>, 
+        clock: &Clock,
+        item_id: ID,
+        _ctx: &mut TxContext)
+```
+
+## Sequence diagrams
```

Comment:
For the various flows, can you also describe various unhappy paths, and how the protocol prevents them?

Reply:
Not sure how this could be included along with the sequence diagrams but I tried to include some cases that could be prevented as comments on the bodies of the functions that I added.